zkchannels-crypto/examples/equality_proofs.rs

```diff
@@ -0,0 +1,173 @@
+use bls12_381::{G1Projective, Scalar};
+use ff::Field;
+use zkchannels_crypto::{
+    pedersen::PedersenParameters,
+    pointcheval_sanders::{KeyPair, PublicKey},
+    proofs::{
+        ChallengeBuilder, CommitmentProof, CommitmentProofBuilder, SignatureProof,
+        SignatureProofBuilder,
+    },
+    Message, Rng,
+};
+
+/// Zero knowledge proof of knowledge of a message tuple that repeats itself
+/// e.g. (x, x).
+pub struct RepeatedMessageProof {
+    proof: CommitmentProof<G1Projective, 2>,
+}
+
+impl RepeatedMessageProof {
+    pub fn new(
+        rng: &mut impl Rng,
+        pedersen_parameters: &PedersenParameters<G1Projective, 2>,
+        secret_value: u64,
```

Comment:
I did this intentionally so that the proof APIs would be on normal types rather than BLS12-381 representations.

Reply:
OK, but I think we need to surface the message encoding. It's something you need to think about when using this library.

Reply:
That is fair. I updated the APIs to use `Scalar`s in most places, but do you think I should add some explicit documentation about `Message`s and `Scalar`s?

Reply:
I think it may be nice to have at least a sentence somewhere that to the effect that proper use requires mapping whatever your domain of interest is to the `Scalars` in an unambiguous way, yeah.